core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -530,12 +533,15 @@ ClientStream newSubstream(ClientStreamTracer.Factory tracerFactory, Metadata new
       ObjectPool<? extends Executor> balancerRpcExecutorPool,
       Supplier<Stopwatch> stopwatchSupplier,
       List<ClientInterceptor> interceptors,
-      final TimeProvider timeProvider) {
+      final TimeProvider timeProvider,
+      ProxyDetector proxyDetector) {
     this.target = checkNotNull(builder.target, "target");
     this.nameResolverFactory = builder.getNameResolverFactory();
     this.nameResolverParams = checkNotNull(builder.getNameResolverParams(), "nameResolverParams");
-    this.nameResolver = getNameResolver(target, nameResolverFactory, nameResolverParams);
+    this.nameResolver = getNameResolver(target, nameResolverFactory, nameResolverParams,
+        proxyDetector);
     this.timeProvider = checkNotNull(timeProvider, "timeProvider");
+    this.proxyDetector = proxyDetector;
```

Comment:
This should be `builder.proxyDetector`. We "simplified" it to avoid passing all the parameters individually. That would also mean you would revert some of the changes to the tests.